Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp

```diff
@@ -1443,12 +1459,15 @@ ErrorOr<Optional<ReadonlyBytes>> PNGImageDecoderPlugin::icc_data()
 
     if (m_context->embedded_icc_profile.has_value()) {
         if (!m_context->decompressed_icc_profile.has_value()) {
-            auto result = Compress::ZlibDecompressor::decompress_all(m_context->embedded_icc_profile->compressed_data);
-            if (!result.has_value()) {
-                m_context->embedded_icc_profile.clear();
-                return Error::from_string_literal("PNGImageDecoderPlugin: Decompression of ICC profile failed");
-            }
-            m_context->decompressed_icc_profile = move(*result);
+            auto compressed_data_stream = make<FixedMemoryStream>(m_context->embedded_icc_profile->compressed_data);
+            auto decompressor_or_error = Compress::ZlibDecompressor::create(move(compressed_data_stream));
```

Comment:
Pretty sure it's the same reason that forced me to not mention these on the first review.